configs/125M-moe.yml

```diff
@@ -0,0 +1,99 @@
+# GPT-2 pretraining setup
+{
+   # Have 4 experts per layer (every 2 layers by default)
```

Comment:
Let's explicitly show people which layers are experts in this example config.

Reply:
Also, is there a specific reason to have it be "every x layers" instead of letting the user specify which layers it is precisely?

I think it makes sense to model this argument after `attention_config`.

Reply:
I agree that this would be more flexible, but I advocate for splitting this "per-layer control" support into a separate PR. 

Now that we're adding support for RWKV/Mamba/MoE, I suspect more users will want to combine blocks in new ways (e.g. attn --> mamba --> MoE ...). Doing this all at once instead of having another MoE-specific per-block flag will save us work in the long run.

In short, I think this PR is fine as-is and per-layer control should be separated into another PR.